src/NServiceBus.FileBasedRouting/XmlRoutingFileParser.cs

```diff
@@ -42,23 +45,32 @@ public IEnumerable<EndpointRoutingConfiguration> Parse(XDocument document)
             return configs;
         }
 
-        static IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
+        IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
         {
             var handles = endpointElement.Element("handles");
 
             var separatelyConfiguredCommands = handles
                 ?.Elements(singular)
                 .Select(e => SelectCommand(e.Attribute("type").Value))
+                .Where(e => e != null)
                 .ToArray() ?? Type.EmptyTypes;
 
             var filteredCommands = handles?.Elements(plural).SelectMany(SelectMessages) ?? Type.EmptyTypes;
             var allCommands = separatelyConfiguredCommands.Concat(filteredCommands).Distinct();
             return allCommands;
         }
 
-        static Type SelectCommand(string typeName)
+        Type SelectCommand(string typeName)
         {
-            return Type.GetType(typeName, true);
+            try
+            {
+                return Type.GetType(typeName, true);
+            }
+            catch
+            {
+                logger.Warn($"Could not find type {typeName}.");
```

Comment:
should we improve that log statement? In many cases, the GetType call shouldn't throw but just return null. In that case we don't log while if theres some exception, it's probably not about the type not being found, but something else. Maybe something like:
```
var type;
try
{
    type = Type.GetType(...)
} 
catch
{
    // ignore 
}

if(type == null)
{
    logger.Warn("cannot add route for unknown type ...");
}

return type;
```